### PR TITLE
[backend] Adding a new endpoint to just test a mapper

### PR DESCRIPTION
**Pull request by Dimfacion**
### Proposed changes

* Adding a new endpoint to test a mapper without saving it first
*

### Related issues

*
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by RomuDeuxfois — APPROVED** [review]

File: openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java [hunk 1]
```diff
@@ -34,14 +43,30 @@ public ImportPostSummary importXLSFile(@PathVariable @NotBlank final String scen
         return injectService.storeXlsFileForImport(file);
     }
 
+    @PostMapping(SCENARIO_URI + "/{scenarioId}/xls/{importId}/dry")
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java:46]: This one is the /test

File: openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java [hunk 2]
```diff
@@ -34,14 +43,30 @@ public ImportPostSummary importXLSFile(@PathVariable @NotBlank final String scen
...
+                                               @Valid @RequestBody final InjectsImportInput input) {
+        // Getting the mapper to use
+        ImportMapper importMapper = importMapperRepository
+                .findById(UUID.fromString(input.getImportMapperId()))
+                .orElseThrow(() -> new ElementNotFoundException(String.format("The import mapper %s was not found", input.getImportMapperId())));
+
+        return injectService.importInjectIntoScenarioFromXLS(scenarioId, importMapper, importId, input.getName(), input.getTimezoneOffset(), false);
+    }
+
     @PostMapping(SCENARIO_URI + "/{scenarioId}/xls/{importId}/test")
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java:61]: And this one the dry run no ?

File: openbas-api/src/main/java/io/openbas/rest/scenario/form/InjectsImportTestInput.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.openbas.rest.mapper.form.ImportMapperAddInput;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+import static io.openbas.config.AppConfig.MANDATORY_MESSAGE;
+
+@Data
+public class InjectsImportTestInput {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/scenario/form/InjectsImportTestInput.java:12]: Why this name ?
Why we don't keep the nomenclature XlsImportMapperTestInput

Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/scenario/form/InjectsImportTestInput.java:12]: The previous POJO in io/openbas/rest/scenario/form was InjectsImportInput so we're keeping the nomenclature used for the other ones here (or should we rename them both ?)

File: openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java (hunk 2 above)
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java:61]: The way I see it is : 
/dry when you want to do a dry run right before importing for good
/test when you want to test the mapper itself
I'm fine with changing the names though if needed

Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java:61]: It's ok for me

File: openbas-api/src/main/java/io/openbas/rest/scenario/form/InjectsImportTestInput.java (hunk 3 above)
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/scenario/form/InjectsImportTestInput.java:12]: I think I'm having difficulty with the whole nomenclature
We can talk about it in person

File: openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java [hunk 4]
```diff
@@ -34,14 +43,30 @@ public ImportPostSummary importXLSFile(@PathVariable @NotBlank final String scen
...
     @Transactional(rollbackOn = Exception.class)
     @Operation(summary = "Test the import of injects from an xls file")
     @Secured(ROLE_USER)
     public ImportTestSummary testImportXLSFile(@PathVariable @NotBlank final String scenarioId,
                                                @PathVariable @NotBlank final String importId,
-                                               @Valid @RequestBody final InjectsImportInput input) {
-        return injectService.importInjectIntoScenarioFromXLS(scenarioId, input, importId, false);
+                                               @Valid @RequestBody final InjectsImportTestInput input) {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java:67]: should be in admin side

Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java]: Done

File: openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java [hunk 5]
```diff
@@ -25,23 +32,24 @@
...
-        return injectService.storeXlsFileForImport(file);
-    }
-
-    @PostMapping(SCENARIO_URI + "/{scenarioId}/xls/{importId}/test")
+    @PostMapping(SCENARIO_URI + "/{scenarioId}/xls/{importId}/dry")
     @Transactional(rollbackOn = Exception.class)
     @Operation(summary = "Test the import of injects from an xls file")
     @Secured(ROLE_USER)
-    public ImportTestSummary testImportXLSFile(@PathVariable @NotBlank final String scenarioId,
+    public ImportTestSummary dryRunImportXLSFile(@PathVariable @NotBlank final String scenarioId,
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/scenario/ScenarioImportApi.java:42]: What's the difference between this endpoint and the next one ?

**Pull request merged**